src/java.base/share/classes/jdk/internal/classfile/snippet-files/PackageSnippets.java

```diff
@@ -0,0 +1,306 @@
+        // @start region="gatherDependencies2"
+        ClassModel cm = Classfile.parse(bytes);
+        Set<ClassDesc> dependencies =
+              cm.elementStream()
+                .flatMap(ce -> ce instanceof MethodMethod mm ? mm.elementStream() : Stream.empty())
+                .flatMap(me -> me instanceof CodeModel com ? com.elementStream() : Stream.empty())
+                .<ClassDesc>mapMulti((xe, c) -> {
+                    switch (xe) {
+                        case InvokeInstruction i -> c.accept(i.owner().asSymbol());
+                        case FieldInstruction i -> c.accept(i.owner().asSymbol());
+                        default -> { }
+                    }
+                })
+                .collect(toSet());
+        // @end
+    }
+
+    void writeHelloWorld() {
+        // @start region="helloWorld"
+        byte[] bytes = Classfile.build(ClassDesc.of("Hello"), cb -> {
+            cb.withFlags(AccessFlag.PUBLIC);
+            cb.withMethod("<init>", MethodTypeDesc.of(ConstantDescs.CD_void), Classfile.ACC_PUBLIC,
+                          mb -> mb.withCode(
+                                  b -> b.aload(0)
+                                        .invokespecial(ConstantDescs.CD_Object, "<init>",
+                                                       MethodTypeDesc.of(ConstantDescs.CD_void))
+                                        .returnInstruction(TypeKind.VoidType)
+                          )
+              )
+              .withMethod("main", MethodTypeDesc.of(ConstantDescs.CD_void, ConstantDescs.CD_String.arrayType()),
+                          Classfile.ACC_PUBLIC,
+                          mb -> mb.withFlags(AccessFlag.STATIC, AccessFlag.PUBLIC)
+                                  .withCode(
+                                  b -> b.getstatic(ClassDesc.of("java.lang.System"), "out", ClassDesc.of("java.io.PrintStream"))
+                                        .constantInstruction(Opcode.LDC, "Hello World")
+                                        .invokevirtual(ClassDesc.of("java.io.PrintStream"), "println",
+                                                       MethodTypeDesc.of(ConstantDescs.CD_void, ConstantDescs.CD_String))
+                                        .returnInstruction(TypeKind.VoidType)
+            ));
+        });
+        // @end
+    }
+
+    void stripDebugMethods1(byte[] bytes) {
+        // @start region="stripDebugMethods1"
+        ClassModel classModel = Classfile.parse(bytes);
+        byte[] newBytes = Classfile.build(classModel.thisClass().asSymbol(),
+                                          classBuilder -> {
+                                              for (ClassElement ce : classModel) {
+                                                  if (!(ce instanceof MethodModel mm
+                                                        && mm.methodName().stringValue().startsWith("debug")))
+                                                  classBuilder.with(ce);
```

Comment:
Indentation is confusing (even though its less concise its probably better to use braces for all the examples):
```suggestion
                                                      classBuilder.with(ce);
```